### ISSUE TITLE
OS.alert() should refocus back to Scene/Control it came from. Do not lose keyboard inputs

### ISSUE THREAD

**Issue by wyattbiker** [godotengine/godot#22514]
**Godot version:**
3.1 alpha 
![screen shot 2018-09-27 at 1 26 39 pm](https://user-images.githubusercontent.com/930478/46213609-fd330700-c306-11e8-864a-9acda2b00214.png)
Mac OS X, High Sierra

**Issue description:**

- OS.alert() loses focus, requiring a second mouse click on the underlying Scene. This means after you click the alert "OK" button, you have to click twice on the Scene for any control to react.
- Also, Keyboard input is lost once the alert is closed.  Example: Press space bar on check box. You will not be able to use any keys again after you close alert.

```
extends CheckBox

func _ready():
	self.grab_focus()
	pass

func _on_CheckBox_toggled(button_pressed):
	OS.alert("Hey")
	pass # Replace with function body.
```

**Comment by TheCire** [comment]
I can confirm that this doesn't work on macOS Mojave too

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in [Godot 3.2.2](https://godotengine.org/article/maintenance-release-godot-3-2-2) or any later release?

If yes, please ensure that an up-to-date Minimal Reproduction Project (MRP) is included in this report (a MRP is a zipped Godot project with the minimal elements necessary to reliably trigger the bug). You can upload ZIP files in an issue comment with a drag and drop.

**Comment by wyattbiker** [comment]
Just tested it with 3.2.1. It is 1/2 fixed.  The 2nd part of the issue, which means you can use the keyboard fine after you close the dialog. This was the important one I guess.

The first issue still happens, but can be confusing. After clicking the OK to make alert go away, you need to click twice on any control to gain focus.

**Comment by wyattbiker** [comment]
Attaching MRP zip file.
[OSAlertBugMRP.zip](https://github.com/godotengine/godot/files/4854831/OSAlertBugMRP.zip)

**Comment by asmaloney** [comment]
More data (on macOS 10.14.6):

**Godot Engine v3.2.3.rc.mono.custom_build.a2edf04fd**
OpenGL ES 3.0 Renderer: AMD Radeon Pro 580X OpenGL Engine

When I run it and click in "Line Edit 1", then click the "Click For Alert" button:
- dialog shows up
- click "OK"
- the game window is no longer active, so it does not receive any input
- if I click the window's title bar to active it, the focus in the line edit has been lost

--

**Godot Engine v4.0.dev.custom_build.dc456059a**

Error `Invalid call. Nonexistent function 'alert' in base '_OS'.`. I guess this is either not re-implemented in 4.0 yet or it's being replaced with something?
